src/test/java/org/jenkinsci/plugins/workflow/cps/CpsThreadTest.java

```diff
@@ -67,7 +67,8 @@
         List<CauseOfInterruption> causes = iba.getCauses();
         assertEquals(1, causes.size());
         assertEquals(CauseOfInterruption.UserInterruption.class, causes.get(0).getClass());
-        r.assertLogContains("never going to stop", b); // TODO under heavy load sometimes fails here; why?
+        r.waitForMessage("Finished: ABORTED", b); // TODO WorkflowRun.isBuilding() can go to false before .finish has completed
```

Comment:
Scratch that, I am not really sure I know why this flakes out. `waitForCompletion` should have been waiting for `completed` to go to `true`, which should only happen after the log is closed. Perhaps there is some race condition in the log writing vs. reading.

At any rate, `waitForMessage` is useful here since the `Finished` message is printed after the `AbortException.message`.